repondeur/zam_repondeur/services/fetch/http.py

```diff
@@ -1,16 +1,29 @@
-import os
-from pathlib import Path
+from typing import Optional
 
 import requests
 from cachecontrol import CacheControl
 from cachecontrol.caches.file_cache import FileCache
+from pyramid.config import Configurator
+from pyramid.registry import Registry
+from pyramid.threadlocal import get_current_registry
+from zope.interface import Interface
 
-HERE = Path(__file__)
-DEFAULT_HTTP_CACHE_DIR = HERE.parent.parent.parent / ".web_cache"
 
-HTTP_CACHE_DIR = os.environ.get("ZAM_HTTP_CACHE_DIR", DEFAULT_HTTP_CACHE_DIR)
+class IHTTPSession(Interface):
+    pass
 
 
-session = requests.session()
+def includeme(config: Configurator) -> None:
+    """
+    Called automatically via config.include("zam_repondeur.services.fetch.http")
+    """
+    session = requests.session()
+    http_cache_dir = config.registry.settings["zam.http_cache_dir"]
+    cached_session = CacheControl(session, cache=FileCache(http_cache_dir))
+    config.registry.registerUtility(component=cached_session, provided=IHTTPSession)
 
-cached_session = CacheControl(session, cache=FileCache(HTTP_CACHE_DIR))
+
+def get_http_session(registry: Optional[Registry] = None) -> CacheControl:
+    if registry is None:
+        registry = get_current_registry()
+    return registry.queryUtility(IHTTPSession)
```

Comment:
What if we add `cached_session = get_http_session()` at the end of the file? I guess we don't want to instanciate it this early?

Reply:
Exactly, we don't want to initialize it at import time (too early). We want to wait for the app to read its config and call the `includeme()` for initialization.

Reply:
OK